wgpu-core/src/binding_model.rs

```diff
@@ -249,6 +255,102 @@ pub struct PipelineLayout<B: hal::Backend> {
     pub(crate) device_id: Stored<DeviceId>,
     pub(crate) life_guard: LifeGuard,
     pub(crate) bind_group_layout_ids: ArrayVec<[Stored<BindGroupLayoutId>; MAX_BIND_GROUPS]>,
+    pub(crate) push_constant_ranges: ArrayVec<[wgt::PushConstantRange; wgt::SHADER_STAGE_COUNT]>,
+}
+
+impl<B: hal::Backend> PipelineLayout<B> {
+    /// Validate push constants match up with expected ranges.
+    pub(crate) fn validate_push_constant_ranges(
+        &self,
+        stages: wgt::ShaderStage,
+        offset: u32,
```

Comment:
So this is basically the same data as presented from RenderCommand::SetPushConstants, which, because it is copy and Range<u32> is not, cannot have a range. It seems silly to construct a range from offset + size when that's what we already have.